### PR TITLE
Installation persian fixed

### LINKED ISSUE #11354: Can't install 1.7.5.0 in Persian
**Describe the bug**
Can't install 1.7.5.0 in Persian

**To Reproduce**
1. Try to install Prestashop 1.7.5.0 in Persian

It was workig in 1.7.4.3, maybe it's MBO ?

**Screenshots**
capture d ecran_602

**Additionnal information**
PrestaShop version: N/A
PHP version: N/A

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | The installation process blocked when installing ps_mbo, this happened because the Tab `Module Catalog` didn't have any translation in persians, thus the validation stated that the field name was empty. To patch this issue during the installation process when data is injected in database if no translation is found for an entity field then we fallback on the english value. On the side a change was made to avoid using cookie during the installation to get the language, and a better management of errors during module installation.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #11354
| How to test?  | Clone the branch and go to the install-dev address, try to install PrestaShop in persian and go to the end of the process.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11441)

### CODE REVIEW

File: src/PrestaShopBundle/Install/XmlLoader.php [hunk 1]
```diff
@@ -309,7 +315,16 @@ public function populateEntity($entity)
                         continue;
                     }
 
-                    if (($node_lang = $xml_lang->xpath($xpath_query)) || ($node_lang = $xml_langs[$default_lang]->xpath($xpath_query))) {
+                    //Triple safety, first requested language, then default one, finally fallback language (en)
+                    $node_lang = $xml_lang->xpath($xpath_query);
+                    if (!$node_lang && $default_lang && isset($xml_langs[$default_lang])) {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Install/XmlLoader.php:318]:
```
foreach ([$default_lang, $fallback_lang] as $lang) {
    if (isset($xml_langs[$lang]) && $node_lang = $xml_langs[$lang]->xpath($xpath_query)) {
        break;
    }
}
```
or something like to remove duplicate

File: src/PrestaShopBundle/Install/Install.php [hunk 2]
```diff
@@ -1049,10 +1049,21 @@ public function installModules($module = null)
...
+            $moduleException = null;
+            try {
+                $moduleInstalled = $moduleManager->install($module_name);
+            } catch (\PrestaShopException $e) {
+                $moduleInstalled = false;
+                $moduleException = $e->getMessage();
+            }
+
+            if (!$moduleInstalled) {
                 /*$module_errors = $module->getErrors();
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Install/Install.php:1063]: can you remove commented code? :)

**Review by mickaelandrieu — COMMENTED** [review]
some minor comments to address, good job

File: src/PrestaShopBundle/Install/Install.php [hunk 3]
```diff
@@ -449,6 +450,7 @@ public function installDefaultData($shop_name, $iso_country = false, $all_langua
                         }
                     }
                 }
+                $iso_codes_to_install[] = self::FALLBACK_LANGUAGE;
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Install/Install.php:453]: As I said, always installing English is a bad idea. All our merchants do not need this language to be installed and may remove it anyway.

Comment (jolelievre) [reply src/PrestaShopBundle/Install/Install.php:453]: ok I will drop this modification, and get back the required changes in the other PR (which removes useless xml files)

File: src/PrestaShopBundle/Install/XmlLoader.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Install/XmlLoader.php:318]: this is gonna be removed

**Comment by jolelievre** [comment]
@Quetzacoalt91 @mickaelandrieu @PierreRambaud This PR is gonna be closed and is therefore replaced by this one:
https://github.com/PrestaShop/PrestaShop/pull/11448

The support of a forced fallback language is dropped.

**Pull request closed without merging**